### ISSUE TITLE
function "obtain_direction " is not found

### ISSUE THREAD

**Issue by aohenuo** [shengliu66/ICV#8]
In the definition of the `task_agent` class in your code, theoretically, there should be an `obtain_direction` to get the insertion position for the MLP, but I did not find its implementation in the class definition. Is this a bug?

**Comment by RadiantCrystal** [comment]
Same issue I am facing. Hello Authors, can you please take care this issue?

**Comment by sumingfirst** [comment]
你的邮件我受到非常感谢配合

**Comment by RadiantCrystal** [comment]
Authors, Can you please upload the missing function in proper place? Also, I am confused about the direction taken in code. Can you help me to understand the following code segment from your forward function of AdapterLayer class? I am not able to map to toxicity reduction use case.

`for i in range(len(self.icvs)):
              lambda_sim = 1.0 + torch.max(torch.tensor([0.]).to(x.device), F.cosine_similarity(x.float(), self.icvs[i][None,None,:], dim=-1)).unsqueeze(-1)
                icv_all_tasks -= alpha[i] * lambda_sim * F.normalize(self.icvs[i], dim=-1).repeat(1,x.shape[1],1)
            icv_all_tasks = 0.1 * icv_all_tasks/len(self.icvs)
            x = F.normalize(F.normalize(x.float(),dim=-1) +  icv_all_tasks, dim=-1) * norm`

**Comment by shengliu66** [comment]
Sorry for the late response, we update the new version of the code. Should work now.
